pop-cli-for-smart-contracts/guides/migrating-to-inkv6.md

```diff
@@ -0,0 +1,72 @@
+---
+description: This guide will walk you through getting started with ink! v6.
+---
+
+# Getting Started with ink! v6
+
+Pop CLI version `v0.7.0` introduces experimental support for ink! v6 smart contracts running on [PolkaVM (RISC-V)](https://github.com/paritytech/polkavm) via `pallet-revive`. This guide helps you transition from ink! v5 (WebAssembly) and start experimenting with ink! v6.
+
+### What's new in ink! v6?
+- **ink! v5**: Version of the ink! that can be compiled into WebAssembly byte code and can be executed on [`pallet-contracts`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/contracts) virtual machine.
+- **ink! v6**: Version of ink! that can be compiled into RISC-V byte code and can be executed on PolkaVM ([`pallet-revive`](https://github.com/paritytech/polkadot-sdk/tree/master/substrate/frame/revive)).
+
+For more information about the benefits of RISC-V and `pallet-revive`, see [RISC-V + pallet-revive](https://use.ink/6.x/current-state#-risc-v--pallet-revive-2025).
+
+### Installing Pop CLI for ink! v6 support
+
+Pop CLI supports ink! v6 through the `polkavm-contracts` feature flag. Install it with:
+
+```
+cargo install --force --locked pop-cli --no-default-features -F polkavm-contracts,parachain
+```
+
+> **⚠️ Note:** Make sure you're using `Rust 1.85` or higher, older versions will fail to compile. You can check your version with rustc `--version`.
+
+### Getting started with ink! v6
+
+The workflow for developing smart contracts remains similar to ink! v5. To get started, see our guide:
+[Create a new contract](./create-a-new-contract.md).
+
+
+#### Migrating existing ink! smart contracts
+If you already have an ink! smart contract, update your dependencies as follows:
+Update your `Cargo.toml`
+
+```toml
+[dependencies]
+ink = { version = "6.0.0-alpha", default-features = false }
+
+[dev-dependencies]
+ink_e2e = { version = "6.0.0-alpha" }
+```
+
+For a complete example, check the [ink! flipper example](https://github.com/use-ink/ink-examples/tree/v6.x/flipper).
+
+#### Account Mapping
+`pallet_revive` enables the deployment and execution of PolkaVM smart contracts. It introduces support for 20-byte accounts (like `EVM`), while most Polkadot-based chains use 32-byte accounts (`AccountId32`). To ensure compatibility, `pallet_revive` provides a mapping feature that lets 32-byte accounts seamlessly interact with the VM.
+
+When interacting with `pallet_revive` for the first time, you’ll need to map your account, this registers your Polkadot account-ID (32-byte address) to a corresponding Ethereum compatible account id (20-byte address), enabling it to interact with contracts.
```

Comment:
sentence feels a little long. Here is an AI recommendation to be a little more concise
```suggestion
When first interacting with `pallet_revive`, you must map your Polkadot account ID (32 bytes) to an Ethereum-compatible 20-byte address. This registration enables your account to interact with contracts
```

Reply:
https://github.com/r0gue-io/pop-docs/pull/28/commits/2121d413f459480ef60bc1ec1ca96bafb46c17ba